### PR TITLE
Feature/add swagger spec

### PR DESCRIPTION
**Pull request by Oluwawunmi**
### What

Add swagger spec service

### How to review

Check is ok.

### Who can review

Anyone.

### CODE REVIEW

File: specification.yml [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+              previous_date:
+                type: string
+        provisional_date:
+          type: string
+          description: A provisional date string, only used with release data_type
+          example: October-November 2024
+      required:
+        - uri
+        - title
+        - content_type
```
Comment (cookel2) [inline specification.yml:188]: Please add a new line to the end of the specification (to avoid the git error etc).

**Review by cookel2 — CHANGES_REQUESTED** [review]
Just one small comment. Also, have we switched from using Swagger 2.0 to using Open API instead now then (by the looks of it)?

**Comment by Oluwawunmi** [comment]
> Just one small comment. Also, have we switched from using Swagger 2.0 to using Open API instead now then (by the looks of it)?

Seems like it. However, OpenAPI 3.0 is basically Swagger 3.0.

**Review by cookel2 — APPROVED** [review]
LGTM

**Pull request merged**